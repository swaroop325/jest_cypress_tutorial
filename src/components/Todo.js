import React from "react";

export default function Todo({ todo }) {
  return <h1 data-testid={`todo-${todo.id}`}>{todo.title}</h1>;
}
