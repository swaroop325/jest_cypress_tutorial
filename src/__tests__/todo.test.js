import Todo from "../components/Todo";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";

test("sample", () => {
  const todo = { id: 1, title: "one", completed: false };
  render(<Todo todo={todo} />);
  const todoElem = screen.getByTestId(`todo-${todo.id}`);
  expect(todoElem).toBeInTheDocument();
  expect(todoElem).toHaveTextContent(todo.title);
});

test("match snap", () => {
  const todo = { id: 1, title: "one", completed: false };
  const tree = renderer.create(<Todo todo={todo} />).toJSON();
  expect(tree).toMatchSnapshot();
});
