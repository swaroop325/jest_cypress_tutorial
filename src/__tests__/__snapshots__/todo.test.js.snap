// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snap 1`] = `
<h1
  data-testid="todo-1"
>
  one
</h1>
`;
